tests/environment.py

```diff
@@ -268,16 +271,19 @@ def __init__(self, name: str, device: None = None, is_controller: bool = False):
                           config_file.read()).group('port')
 
         # On WSL, connect to the locally exposed container port
-        if on_wsl:
-            published_port_output = subprocess.check_output(
-                ["docker", "port", name, ucc_port]).decode('utf-8').split(":")
-            device_ip = published_port_output[0]
-            ucc_port = int(published_port_output[1])
+        if not compose:
+            if on_wsl:
+                published_port_output = subprocess.check_output(
+                    ["docker", "port", name, ucc_port]).decode('utf-8').split(":")
+                device_ip = published_port_output[0]
+                ucc_port = int(published_port_output[1])
+            else:
+                device_ip_output = self.command(
+                    'ip', '-f', 'inet', 'addr', 'show', self.bridge_name)
+                device_ip = re.search(
+                    r'inet (?P<ip>[0-9.]+)', device_ip_output.decode('utf-8')).group('ip')
         else:
-            device_ip_output = self.command(
-                'ip', '-f', 'inet', 'addr', 'show', self.bridge_name)
-            device_ip = re.search(r'inet (?P<ip>[0-9.]+)',
-                                  device_ip_output.decode('utf-8')).group('ip')
+            device_ip = self.device.docker_name
```

Comment:
Shouldn't this work in the non-compose version as well? (maybe not on wsl, but on Linux as long as the containers are in the same docker network this should work as well I believe).
(nitpick)

Reply:
Would need to be tested, but as the non-compose code is going to eventually be deleted ...